### PR TITLE
kv: Limit time spent in race-exposing loop

### PR DESCRIPTION
**Pull request by bdarnell**
This test would sleep for up to 300 microseconds between iterations,
which was enough to limit its impact when tests are run on fast
machines, but on slower ones it could lead to a large amount of time
being spent in this loop. Lengthen the sleep based on the time taken by
the previous iteration to ensure that we don't spend more than half our
time here.

May address #9499

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9505)

### CODE REVIEW

**Comment by bdarnell** [comment]
Not as big a difference as I'd hoped - took about 30 seconds off the 7-minute test time.

File: kv/transport_race.go [hunk 1]
```diff
@@ -89,13 +95,20 @@ func grpcTransportFactory(
...
 							panic(err)
 						}
 					}
 				}
+				interval = timeutil.Since(start)
+				// Use defaultRaceInterval as a minimum to limit how much time
+				// we spend here.
+				if interval < defaultRaceInterval {
+					interval = defaultRaceInterval
+				}
```
Comment (vivekmenezes) [inline kv/transport_race.go]: how about moving this into jitter()

**Review by vivekmenezes — APPROVED** [review]
LGTM

File: kv/transport_race.go (hunk 1 above)
Comment (bdarnell) [inline kv/transport_race.go]: Done

**Review by vivekmenezes — APPROVED** [review]
LGTM

Also okay to move the jitter down to the bottom of the loop so you don't need interval

**Comment by bdarnell** [comment]
Good idea. Done.

**Pull request merged**